facdb/geocode/function1B.py

```diff
@@ -0,0 +1,107 @@
+import json
+from functools import wraps
+from multiprocessing import Pool, cpu_count
+
+import pandas as pd
+
+from . import GeosupportError, g
+
+
+class Function1B:
+    def __init__(
+        self,
+        street_name_field: str = None,
+        house_number_field: str = None,
+        borough_field: str = None,
+        zipcode_field: str = None,
+    ):
+        self.street_name_field = street_name_field
+        self.house_number_field = house_number_field
+        self.borough_field = borough_field
+        self.zipcode_field = zipcode_field
+
+    def geocode_a_dataframe(self, df: pd.DataFrame):
+        records = df.to_dict("records")
+        with Pool(processes=cpu_count()) as pool:
+            it = pool.map(self.geocode_one_record, records, 1000)
+        df_geo = pd.DataFrame(it)
+        return df.merge(df_geo, how="left", on="uid", suffixes=("", "_"))
+
+    def geocode_one_record(self, inputs: dict) -> dict:
+        """
+        Note that df needs
+        """
+        uid = inputs.get("uid")
+        input_sname = inputs.get("input_sname")
+        input_hnum = inputs.get("input_hnum")
+        input_borough = inputs.get("input_borough")
+        input_zipcode = inputs.get("input_zipcode")
+        try:
+            geo = g["1B"](
+                street_name=input_sname,
+                house_number=input_hnum,
+                borough=input_borough,
+                zip_code=input_zipcode,
+            )
+        except GeosupportError as e:
+            geo = e.result
+
+        geo = self.parser(geo)
+        return dict(
+            uid=uid,
+            geo_1b=json.dumps(
+                dict(
+                    inputs=dict(
+                        input_sname=input_sname,
+                        input_hnum=input_hnum,
+                        input_borough=input_borough,
+                        input_zipcode=input_zipcode,
+                    ),
+                    result=geo,
+                )
+            ),
+        )
+
+    def parser(self, geo):
+        return dict(
```

Comment:
We should maybe modify the parsers to specify the function used, since we might be using multiple geosupport functions in succession. If we're retaining maximum information, it will be too confusing just having labels of `geo_`. For example, lets change `geo_house_humber` to `house_number_1b`, etc. For 1B in particular, it's helpful to label the GRC and error codes as 1A and 1E, the way we're doing for COLP.